### ISSUE TITLE
Please consider using TypeScript to add type-safety and non-nullables guarantees

### ISSUE THREAD

**Issue by demetriusnunes** [kaleidos/olmo#2]

**Comment by ghost** [comment]
Yes, I know :) I was trying Facebook's Flow but couldn't make it work smoothly, specially when dealing with higher-kinded types. I haven't check TypeScript so I don't know exactly how far can I go with it's type checker but sounds like a good choice to explore. If you are up to it feel free to contribute 😸
